cli/cli.cpp

```diff
@@ -0,0 +1,150 @@
+#include "display.h"
+
+void help() {
+    std::cout << "Rotate and rearrange displays\n";
+    std::cout << "\n";
+    std::cout << "Arguments:\n";
+    std::cout << "\n";
+    std::cout << "  No args     Show this help, same as /?.\n";
+    std::cout << "  /?          Show this help page.\n";
+    std::cout << "  [rotate | /r] [angle] [display]\n";
+    std::cout << "              Rotate the display with the given number clockwise by the given angle.\n";
+    std::cout << "              The angle a must be a multiple of 90 (or 0 to reset any rotation).\n";
+    std::cout << "  [position | /p] [x] [y] [display]\n";
+    std::cout << "              Rearrange the display with the given number to the given coordinates.\n";
+    std::cout << "              Display must be at least 1 and angle a multiple of 90 (or 0 to reset).\n";
+    std::cout << "  [info | /i] [display]\n";
+    std::cout << "              Display information such as name, orientation and position of displays.\n";
+    std::cout << "              If display argument is not speicfied it shows the info for all displays.\n";
+    std::cout << "\n";
+    std::cout << "Display numbers for all arguments start with 1. \n";
+    std::cout << "Unless otherwise specified display argument may be left out with default value of 1.\n";
+    std::cout << "\n";
+}
+
+int rotate(int argc, char** argv) {
+    if (argc < 3) {
+        std::cout << "Missing angle, use argument /? for further help\n";
+        return 1;
+    }
+
+    int disp;
+    if (argc < 4) {
+        disp = { 0 };
+    } else {
+        try {
+            disp = { std::stoi(argv[3]) - 1 };
+        } catch (const std::invalid_argument &ex) {
+            std::cout << "Invalid display number, use argument /? for further help\n";
+            return 1;
+        }
+    }
+    
+    try {
+        Display d{ disp };
+        DisplayRotation rot = static_cast<DisplayRotation>((std::stoi(argv[2]) / 90) % 4);
+        if (rot == cw_0) {
+            return !d.resetRotation();
+        } else {
+            return !d.rotate(rot);
+        }
+    } catch (const std::invalid_argument & ex) {
+        std::cout << "Invalid rotation, use argument /? for further help\n";
+        return 1;
+    } catch (const std::runtime_error & ex) {
+        std::cout << ex.what() << ", use argument /? for further help and /i to see current displays\n";
+        return 1;
+    }
+}
+
+int setPosition(int argc, char** argv) {
+    if (argc < 4) {
+        std::cout << "Missing coords, use argument /? for further help\n";
+        return 1;
+    }
+
+    int disp;
+    if (argc < 5) {
+        disp = { 0 };
+    }
+    else {
+        try {
+            disp = { std::stoi(argv[4]) - 1 };
+        }
+        catch (const std::invalid_argument &ex) {
+            std::cout << "Invalid display number, use argument /? for further help\n";
+            return 1;
+        }
+    }
+
+    try {
+        Display d{ disp };
+        long x{ std::stoi(argv[2]) };
```

Comment:
`std::stoi` only returns an `int` if you really want a long here then you can use `std::stol` instead.